pymeasure/instruments/mksinst/mks937b.py

```diff
@@ -0,0 +1,258 @@
+
+    Connection to the device is made through an RS232/RS485 serial connection.
+    The communication protocol of this device is as follows:
+
+    Query: '@<aaa><Command>?;FF' with the response '@<aaa>ACK<Response>;FF'
+    Set command: '@<aaa><Command>!<parameter>;FF' with the response '@<aaa>ACK<Response>;FF'
+    Above <aaa> is an address from 001 to 254 which can be specified upon
+    initialization. Since ';FF' is not supported by pyvisa as terminator this
+    class overloads the device communication methods.
+
+    :param adapter: pyvisa resource name of the instrument or adapter instance
+    :param address: device address included in every message to the instrument
+                    (default=253)
+    :param kwargs: Any valid key-word argument for Instrument
+    """
+
+    def __init__(self, adapter, name="MKS 937B vacuum gauge controller", address=253, **kwargs):
+        kwargs.setdefault("write_termination", ";FF")
+        kwargs.setdefault("read_termination", ";")  # in reality its ";FF"
+        # which is, however, invalid for pyvisa. Therefore extra bytes have to
+        # be read in the read() method.
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            **kwargs
+        )
+        self.address = address
+        # compiled regular expression for finding numerical values in reply strings
+        self._re_response = re.compile(fr"@{self.address:03d}(?P<ack>ACK)?(?P<msg>.*)")
+
+    def _extract_reply(self, reply):
+        """ preprocess_reply function which tries to extract <Response> from
+        '@<aaa>ACK<Response>;FF'. If <Response> can not be identified the orignal string
+        is returned.
+        :param reply: reply string
+        :returns: string with only the response, or the original string
+        """
+        rvalue = self._re_response.search(reply)
+        if rvalue:
+            return rvalue.group('msg')
+        return reply
+
+    def _prepend_address(self, cmd):
+        """
+        create command string by including the device address
+        """
+        return f"@{self.address:03d}{cmd}"
+
+    def _check_extra_termination(self):
+        """
+        Check the read termination to correspond to the protocol
+        """
+        t = super().read_bytes(2)  # read extra termination chars 'FF'
+        if t != b'FF':
+            raise ValueError(f"unexpected termination string received {t}")
+
+    def read(self):
+        """
+        Reads from the instrument including the correct termination characters
+        """
+        ret = super().read()
+        self._check_extra_termination()
+        return self._extract_reply(ret)
+
+    def write(self, command):
+        """
+        Writes to the instrument including the device address
+
+        :param command: command string to be sent to the instrument
+        """
+        super().write(self._prepend_address(command))
+
+    def check_errors(self):
```

Comment:
It's not clear to me when this method is used.

Reply:
see properties with `check_set_error` set

Reply:
Ok. Thanks for the clarification